### PR TITLE
Add ViT and Image Ranking Support

### PR DESCRIPTION
**Pull request by jacobbieker**
This adds a few features:

1. Support for vision models, primarily ViT-based models, from Hugging Face transformers
2. Support for vision ranking tasks, tested on website quality ranking
3. Scraping code for Wayback machine for creating a training dataset of website images + HTML for training the vision models

The scraper downloads with wget the source files for the wayback machine version, and rewrites the links to be the local paths. This allows for the websites to be loaded in the browser locally without being then limited by Wayback machine's rate limits. The `generate_screenshots.py` then can generate screenshots in Chrome and Firefox in multiple common device resolutions and aspect ratios, and filters out website downloads that did not finish correctly. These screenshots then can be used by the ImageRankingDataset to train the model on the website images for image quality. 

This also fixes a few small bugs, including the created optimizer and scheduler not being passed to the Hugging Face Trainer. It also adds some unit tests.

### CODE REVIEW

File: src_website_quality/vit_model.py [hunk 1]
```diff
@@ -103,10 +107,13 @@ def convert_tensors(self, *tensors):
         return [t.to(dtype=self.config.torch_dtype) if t is not None else None for t in tensors]
 
     def get_pairwise_reward(self, input1, input2, reward1, reward2):
-        end_of_prompt = self.get_end_of_prompt(input1, input2)
-        start_of_padding = max(self.get_start_of_padding(input1), self.get_start_of_padding(input2))
-        return -torch.log(
-            torch.sigmoid(reward1[end_of_prompt:start_of_padding] - reward2[end_of_prompt:start_of_padding])).mean()
+        if self.vision_model:
```
Comment (UYasher) [inline src_website_quality/vit_model.py]: I think some of the logic for this if-statement should be moved up. The transformer case is actually specific to decoder-only models, I think. For encoder models, we could directly get a single score for the reward, I believe. Not necessarily actionable in this PR, but should be updated.

**Comment by jacobbieker** [comment]
Closing this PR, and it has now been broken up into parts:
#4 provides the website scraping and screenshot generation code
#5 provides the PyTorch Dataset that uses those screenshots to generate a ranked dataset
#6 provides the changes to `Model` to support ViT-based image models
#7 provides the updates to the training script to used #5 and #6 together to train the vision models.
